src/OpenTelemetry.Exporter.OpenTelemetryProtocol/IOtlpExporterOptions.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+#if NETFRAMEWORK
+using System.Net.Http;
+#endif
+
+namespace OpenTelemetry.Exporter;
+
+/// <summary>
+/// Describes the OpenTelemetry Protocol (OTLP) exporter options shared by all
+/// signals.
+/// </summary>
+internal interface IOtlpExporterOptions
```

Comment:
Is this interface only being added for DI config purposes? If yes, could you add a comment here similar to the one for `UseOtlpExporterRegistration` [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5400/files#diff-56333e99a46dc269315055dd59f70eeb552bf53512014dda2aa09bb864c02921R8-R10)?

Reply:
Not for DI purposes.

We have this thing: https://github.com/open-telemetry/opentelemetry-dotnet/issues/4527

There are some trace settings on `OtlpExporterOptions`. Generally we (me and @alanwest, maybe others) feel this was a mistake. For metrics & logs we went in a different direction. Those two properties on `OtlpExporterOptions` we want to obsolete. #5423 is kind of the groundwork for that.

This PR is adding `OtlpExporterBuilderOptions` which could have looked like this:

```csharp
internal sealed class OtlpExporterBuilderOptions
{
    public OtlpExporterOptions DefaultOptions { get; }
    public OtlpExporterOptions LoggingOptions { get; }
    public OtlpExporterOptions MetricsOptions { get; }
    public OtlpExporterOptions TracingOptions { get; }
}
```

Problem there is it is using `OtlpExporterOptions` which has the trace settings. What I didn't want to do was carry forward that mistake into this new API. What I did instead was:

```csharp
internal sealed class OtlpExporterBuilderOptions
{
    public IOtlpExporterOptions DefaultOptions { get; }
    public IOtlpExporterOptions LoggingOptions { get; }
    public IOtlpExporterOptions MetricsOptions { get; }
    public IOtlpExporterOptions TracingOptions { get; }
}
```

What `IOtlpExporterOptions` does is expose just the stuff that is _really_ supported by all the signals.

Now this is all internal at the moment. I could remove `IOtlpExporterOptions` from this PR. But I would like to keep it so the contract is there. Hoping to expose some additional overloads to `UseOtlpExporter` to add additional scenarios in the future which will require this stuff to be made public (could be experimental).

Reply:
Thanks for the explanation!